src/Microsoft.AspNetCore.SignalR.Core/Internal/IHubProtocolResolver.cs

```diff
@@ -1,12 +1,13 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System.Collections.Generic;
 using Microsoft.AspNetCore.SignalR.Internal.Protocol;
 
 namespace Microsoft.AspNetCore.SignalR.Internal
 {
     public interface IHubProtocolResolver
     {
-        IHubProtocol GetProtocol(string protocolName, HubConnectionContext connection);
+        IHubProtocol GetProtocol(string protocolName, List<string> supportedProtocols, HubConnectionContext connection);
```

Comment:
`List<T>` in a public API!

![https://media.giphy.com/media/sFTWiBKYYWKVa/giphy.gif](https://media.giphy.com/media/sFTWiBKYYWKVa/giphy.gif)

Use `IReadOnlyList<string>`. The method only needs to be able to read from the list, and doesn't care what kind of list you give it.

Reply:
This should also just be an `IList<string>`, right? It get passed the supportedProtocol list from the HubEndPoint which is an IList

Reply:
Should it not be `List<T>` even if it is in the internal namespace? I wasn't sure about that when I looked at the code.

Reply:
In this case, the API is likely to be made properly public at some point. Also, "pubternal" (public in the Internal namespace) types should still try to keep to our public API standards.